http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -663,23 +703,27 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
 
       auth <- samDAO.getLeoAuthToken
 
-      diskResourceOpt <- controlledResourceQuery
-        .getWsmRecordForRuntime(e.runtimeId, WsmResourceType.AzureDisk)
-        .transaction
-      _ <- diskResourceOpt.traverse { disk =>
-        // TODO: once we start supporting persistent disk, we should not delete disk anymore
-        wsmDao.deleteDisk(
-          DeleteWsmResourceRequest(
-            e.workspaceId,
-            disk.resourceId,
-            DeleteControlledAzureResourceRequest(
-              WsmJobControl(WsmJobId(s"delete-disk-${ctx.traceId.asString.take(10)}"))
+      // if no existing disk being used, delete new disk created
```

Comment:
I am not sure I understand, there should be nothing to delete? It feels contradictory to the TODO comment 
// TODO: once we start supporting persistent disk, we should not delete disk anymore

Reply:
so this TODO was here before my PR, since this is in `handleAzureRuntimeCreationError` basically the current behavior is if it errors during creation, delete the disk it created while creating the runtime if its exists, but with persistent disks we don't want to delete the persistent disk. It grabs from the CONTROLLED_RESOURCE table to decide which disk to delete, which is updated with the new runtimeId when using an existing disk, so the traverse may execute when using an existing disk as well

Reply:
Ahhh, that makes a lot of sense, thanks for explaining!